lib/src/eth.rs

```diff
@@ -1,52 +1,97 @@
-use ethers::prelude::Provider;
-use ethers::types::{Filter, Log};
-use ethers_providers::Ws;
+use alloy_rpc_types::pubsub::{Params, SubscriptionKind, SubscriptionResult};
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
-use tokio::task::JoinHandle;
 
-use crate::core::ProcessId;
-
-/// The Request type that can be made to eth:distro:sys. Currently primitive, this
-/// enum will expand to support more actions in the future.
+/// The Action and Request type that can be made to eth:distro:sys.
 ///
 /// Will be serialized and deserialized using `serde_json::to_vec` and `serde_json::from_slice`.
 #[derive(Debug, Serialize, Deserialize)]
 pub enum EthAction {
     /// Subscribe to logs with a custom filter. ID is to be used to unsubscribe.
-    SubscribeLogs { sub_id: u64, filter: Filter },
+    /// Logs come in as alloy_rpc_types::pubsub::SubscriptionResults
+    SubscribeLogs {
+        sub_id: u64,
+        kind: SubscriptionKind,
+        params: Params,
+    },
     /// Kill a SubscribeLogs subscription of a given ID, to stop getting updates.
     UnsubscribeLogs(u64),
+    /// Raw request. Used by kinode_process_lib.
+    Request {
+        method: String,
+        params: serde_json::Value,
+    },
+}
+/// Incoming subscription update.
```

Comment:
add a whitespace here, and say that this is the incoming Request type that processes will receive from eth. Same change in process_lib PR